src/main/scala/edu/berkeley/ce/rockslicing/JointGenerator.scala

```diff
@@ -0,0 +1,221 @@
+    } else {
+      val dipVector = DenseVector[Double](math.cos(dipRadians) * math.cos(-(strikeRadians + math.Pi / 2.0)),
+        math.cos(dipRadians) * math.sin(-(strikeRadians + math.Pi / 2.0)),
+        -math.sin(dipRadians))
+      val jointNormal = linalg.normalize(linalg.cross(strikeVector, dipVector))
+      Array(jointNormal(0), jointNormal(1), jointNormal(2))
+    }
+  }
+
+  /**
+    * Finds the faces that define the rock volume of interest
+    *
+    * @param globalOrigin Coordinates of global origin
+    * @param rockVolumeData Array of arrays describing each of the faces that define the boundaries of
+    *                       the rock volume of interest. The inputs in each array are strike, dip and the x,
+    *                       y and z coordinates of a point that lies within the plane containing the face.
+    *                       Strike and dip should be specified in degrees.
+    * @return A sequence of faces that represent the rock volume of interest
+    */
+  private def findBoundingFaces(globalOrigin: Array[Double], rockVolumeData: Seq[Array[Double]]): Seq[Face] = {
+    rockVolumeData map { parameters =>
+      val faceCenter = Array(parameters(2), parameters(3), parameters(4))
+      val normal = findJointNormal(parameters(0), parameters(1))
+      val distance = Joint.findDistance(normal, globalOrigin, faceCenter)
+      if (distance > 0.0) {
+        Face(normal, distance, parameters(5), parameters(6))
+      } else {
+        // Ensures that all normals are outward pointing
+        Face(Array(-normal(0), -normal(1), -normal(2)), -distance, parameters(5), parameters(6))
+      }
+    }
+  }
+
+  /**
+    * Finds representative joint for each joint set, centered at the lower left corner of the
+    * bounding box.
+    *
+    * @param globalOrigin Coordinates of the global origin
+    * @param lowerLeftCorner Lower left corner of the bounding box
+    * @param joints Array containing the input data representing joint sets
+    * @return Seq of "master" joints that represent input joint sets. These will be used as a starting point
+    *         to generate full joint set.
+    */
+  private def findMasterJoints(globalOrigin: Array[Double], lowerLeftCorner: Array[Double],
+                               joints: Seq[Array[Double]]): Option[Seq[Joint]] = {
+    val masterJoints = joints flatMap { parameters =>
+      if (parameters.length == 6) {
+        // TODO: Implement non-persistent joint generator
+        val normal = findJointNormal(parameters(0), parameters(1))
+        Some(Joint(normal, globalOrigin, center=Array(lowerLeftCorner(0), lowerLeftCorner(1), lowerLeftCorner(2)),
+          phi=parameters(4), cohesion=parameters(5), shape=Vector.empty))
+      } else {
+        // TODO: Implement stochastic generator
+        println("Error, Joint Input Data: Stochastic joint generation not implemented yet")
+        None
+      }
+    }
+    if (masterJoints.length != joints.length) {
```

Comment:
The option was there to catch if the user had input parameters for stochastic joint generation, but that's taken care of by an `UnsupportedOpertationException` now so I'll get rid of the option on this.